pghoard/transfer.py

```diff
@@ -513,6 +538,10 @@ def handle_upload(self, site, key, file_to_transfer: UploadEvent):
 
             # Sleep for a bit to avoid busy looping. Increase sleep time if the op fails multiple times
             self.sleep(min(0.5 * 2 ** (file_to_transfer.retry_number - 1), 20))
-
+            if file_to_transfer.file_type in (
+                FileType.Basebackup, FileType.Basebackup_chunk, FileType.Basebackup_delta, FileType.Basebackup_delta_chunk
+            ):
+                persisted_progress = PersistedProgress.read(metrics=self.metrics)
+                persisted_progress.reset(key, metrics=self.metrics)
```

Comment:
Should we be resetting the progress to zero on a retry? 
What happens if we are stuck in a reset loop? Technically the backup is stalled but we won't know cause we keep resetting the value.

Reply:
should we limit the reset based on retry?

Reply:
yeah how many times do we currently retry?

Reply:
less than 2

Reply:
an when we don't reset the stalled metric will continue to increase?

Reply:
yep, it will continue to increase, is it better to reset or delete the file when we request a new backup?

Reply:
We should only reset the statistics when we complete the backup successfully. The OG ticket described what to do in this situation as:

> if current progress <= persisted progress, emit a metric (e.g. basebackup-stalled) whose value is the number of seconds since the persisted epoch.

So basically, we should only ever reset stats on success, otherwise emit a metric to indicate we are stalled and no longer making progress. This should prevent missing the endless retry case.